third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc

```diff
@@ -2860,33 +2871,57 @@ Object.defineProperty(window, "__REACT_DEVTOOLS_GLOBAL_HOOK__", {
 let uidCounter = 0;
 
 function inject(renderer) {
+  // Declare these enum strings in scope for later routine use
+  const annotationType = "inject";
+
   const id = ++uidCounter;
-  window.__RECORD_REPLAY_ANNOTATION_HOOK__("react-devtools-hook", "inject");
+  window.__RECORD_REPLAY_ANNOTATION_HOOK__("react-devtools-hook", JSON.stringify({type: annotationType}));
   window.__REACT_DEVTOOLS_SAVED_RENDERERS__.push(renderer);
   return id;
 }
 
 function onCommitFiberUnmount(rendererID, fiber) {
-  window.__RECORD_REPLAY_ANNOTATION_HOOK__("react-devtools-hook", "commit-fiber-unmount");
+  const annotationType = "commit-fiber-unmount"
+
+  // Unmounts are always one fiber at a time during the commit phase.
+  // Stash the unmounted fibers here, so we can map them to persistent 
+  // object IDs inside of `onCommitFiberRoot` processing in the routine.
+  const unmountedFibersSet = getUnmountedFibersSetForRenderer(rendererID);
+  unmountedFibersSet.add(fiber);
+
+  window.__RECORD_REPLAY_ANNOTATION_HOOK__("react-devtools-hook", JSON.stringify({type: annotationType}));
 }
 
 function onCommitFiberRoot(rendererID, root, priorityLevel) {
-  const mountedRoots = stubGetFiberRoots(rendererID);
+  // The "commit" handler should be the only one the routine needs to do the work as of 2023-05-01.
+  // We capture unmounted fibers in the unmount handler above, and the routine
+  // will process them when we evaluate at the commit annotation point.
+  // The others mostly exist for hypothetical completeness.
+  const annotationType = "commit-fiber-root";
+
+  const mountedRoots = getFiberRootsSetForRenderer(rendererID);
   const current = root.current;
   const isKnownRoot = mountedRoots.has(root);
-  const isUnmounting = current.memoizedState == null || current.memoizedState.element == null; // Keep track of mounted roots so we can hydrate when DevTools connect.
+  // Keep track of mounted roots so we can hydrate when DevTools connect.
+  const isUnmounting = current.memoizedState == null || current.memoizedState.element == null; 
 
   if (!isKnownRoot && !isUnmounting) {
     mountedRoots.add(root);
   } else if (isKnownRoot && isUnmounting) {
     mountedRoots.delete(root);
   }
+
+  // Get these so it's in scope in the routine eval, and we can clear it after the annotation
+  const unmountedFibersSet = getUnmountedFibersSetForRenderer(rendererID);
   
-  window.__RECORD_REPLAY_ANNOTATION_HOOK__("react-devtools-hook", "commit-fiber-root");
+  window.__RECORD_REPLAY_ANNOTATION_HOOK__("react-devtools-hook", JSON.stringify({type: annotationType}));
```

Comment:
How would you feel about having this do
```
window.__RECORD_REPLAY_ANNOTATION_HOOK__("react-devtools-hook:v1:" + annotationType, "");
```
to indicate some basic versioning info in the recordings, and to include the annotation type in the kind itself? That also gives us more explicit namespacing and moves more info into the `kind` and out of `contents`.

My main concern is that as soon as we embed this in JSON, we then have to either manually parse that JSON and query it, so we're limiting ourselves to having to fetch all the points, do some preprocessing, and then send a protocol command, and it seems like it'd be awesome to be able to do
```
runEvaluation({
  pointSelector: {
    kind: "annotations",
    types: [
      "react-devtools-hook:v1:commit-fiber-root", 
    ],
  }
})
```
to run an evaluation without having to fetch all the data to the client first.

Reply:
Sure, let's do that.  Gonna knock out a Passport Tour thing for Jon and then I'll make those changes.